packages/flutter/lib/src/cupertino/search_field.dart

```diff
@@ -439,12 +440,13 @@ class _CupertinoSearchTextFieldState extends State<CupertinoSearchTextField> wit
   }
 
   void _handleScrollNotification(ScrollNotification notification) {
-    if (_maxHeight == null) {
-      _maxHeight ??= context.size?.height;
-    } else if (notification is ScrollUpdateNotification) {
+    if (notification is ScrollUpdateNotification) {
       final double currentHeight = context.size?.height ?? 0.0;
       setState(() {
-        _fadeExtent = _calculateScrollOpacity(currentHeight, _maxHeight!);
+        _fadeExtent = _calculateScrollOpacity(
+          currentHeight,
+          widget.itemSize + math.max(widget.prefixInsets.vertical, widget.suffixInsets.vertical),
```

Comment:
Doesn't look like the calculation is accurate, the documentation says the itemSize can be scaled by the system a11y text scale factor